spec/features/admin/invoice_print_spec.rb

```diff
@@ -0,0 +1,99 @@
+                                variants: [product.variants.first])
+  end
+
+  let(:order) do
+    create(:order_with_totals_and_distribution, user: user, distributor: distributor,
+                                                order_cycle: order_cycle, state: 'complete',
+                                                payment_state: 'balance_due')
+  end
+
+  describe "that contains right Payment Description at Checkout information" do
+    let!(:payment_method1) do
+      create(:stripe_sca_payment_method, distributors: [distributor], description: "description1")
+    end
+    let!(:payment_method2) do
+      create(:stripe_sca_payment_method, distributors: [distributor], description: "description2")
+    end
+
+    context "with no payment" do
+      it "do not display the payment description information" do
+        login_as_admin_and_visit spree.print_admin_order_path(order)
+        convert_pdf_to_page
+        expect(page).to have_no_content 'Payment Description at Checkout'
+      end
+    end
+
+    context "with one payment" do
+      let!(:payment1) do
+        create(:payment, order: order, state: 'completed', payment_method: payment_method1)
+      end
+      before do
+        order.save!
+      end
+
+      it "display the payment description section" do
+        login_as_admin_and_visit spree.print_admin_order_path(order)
+        convert_pdf_to_page
+        expect(page).to have_content 'Payment Description at Checkout'
+        expect(page).to have_content 'description1'
+      end
+    end
+
+    context "with two payments, and one that failed" do
+      before do
+        order.update payments: []
+        order.payments << create(:payment, order: order, state: 'completed', payment_method: payment_method1, created_at: 1.day.ago)
+        order.payments << create(:payment, order: order, state: 'failed', payment_method: payment_method2, created_at: 2.days.ago)
+        order.save!
+      end
+
+      it "display the payment description section and use the one from the completed payment" do
+        login_as_admin_and_visit spree.print_admin_order_path(order)
+        convert_pdf_to_page
+        expect(page).to have_content 'Payment Description at Checkout'
+        expect(page).to have_content 'description1'
+      end
+    end
+
+    context "with two completed payments" do
+      before do
+        order.update payments: []
+        order.payments << create(:payment, order: order, state: 'completed', payment_method: payment_method1, created_at: 2.days.ago)
+        order.payments << create(:payment, order: order, state: 'completed', payment_method: payment_method2, created_at: 1.day.ago)
+        order.save!
+      end
+
+      it "display the payment description section and use the one from the first payment" do
+        login_as_admin_and_visit spree.print_admin_order_path(order)
+        convert_pdf_to_page
+        expect(page).to have_content 'Payment Description at Checkout'
```

Comment:
I don't think there's need for the `instance_variable_set` hack if we store the content in a var and do `expect(file_contents).to have_content('Payment Description at Checkout') right?

Reply:
aaah, I see. Ok then, the hack seems justified :+1: